syft/messaging/plan.py

```diff
@@ -134,16 +214,20 @@ def __init__(
 
         self.tags = None
         self.description = None
-        self.blueprint = blueprint
 
-        # For methods
-        self.is_method = is_method
-        self._self = None
+        if blueprint is not None:
+            self.forward = blueprint
+        elif self.is_built:
+            self.forward = None
+
+    def add_to_state(self, *elements):
+        for elem in elements:
+            self.state.append(elem)
 
     def _auto_build(self, args_shape: List[Tuple[int]] = None):
         placeholders = self._create_placeholders(args_shape)
-        args = [self._self] + placeholders if self.is_method else placeholders
-        self._build(args)
+        args = placeholders
```

Comment:
I think you can get rid of this variable and just call placeholders args instead